### PR TITLE
gh-208: remove legacy random number generation

### PR DESCRIPTION
**Pull request by Saransh-cpp**
Gets rid of legacy `RandomState` and replaced it with the new random number `Generator` for rng calls.

Fixes: #208

### CODE REVIEW

**Review by paddyroddy — CHANGES_REQUESTED** [review]
I think it would be good to have the random seed as a test constant or `pytest.fixture`

**Comment by ntessore** [comment]
> I think it would be good to have the random seed as a test constant or `pytest.fixture`

Agree, I usually make `rng` itself the fixture, which can be helpful for testing with a mock RNG.

File: tests/core/test_algorithm.py [hunk 1]
```diff
@@ -10,15 +10,22 @@
     HAVE_SCIPY = True
 
 
+@pytest.fixture
+def rng():
+    import numpy as np
+
+    return np.random.default_rng(seed=42)
+
+
```
Comment (ntessore) [inline tests/core/test_algorithm.py:19]: Move this to conftest.py so it only needs to be defined once?

**Review by ntessore — CHANGES_REQUESTED** [review]

File: tests/core/test_algorithm.py (hunk 1 above)
Comment (Saransh-cpp) [reply tests/core/test_algorithm.py]: Sounds better, thanks!

**Review by ntessore — APPROVED** [review]

**Pull request merged**